### ISSUE TITLE
tab creation time?

### ISSUE THREAD

**Issue by vu3rdd** [osnr/TabFS#33]
First of all, `tabfs` is great! Thanks for this wonderful project.

I don't know if browsers provide an API to query the tab creation time. It would be nice if the file `ctime` reflects the tab creation time.

I am imagining a situation where a user can do operations like deleting all tabs older than a date or something of that sort..

**Comment by osnr** [comment]
Yeah, I think this would be great. I don't think there is an API for it, although I could be wrong -- I make a note about this [here](https://omar.website/tabfs/#hmm) (added emphasis):

> for a lot of things in the extension API, the browser can notify you of updates but there's no apparent way to query the full current state. so we'd need to sit in a lot of these places from the beginning and accumulate the incoming events to know, like, **the last time a tab was updated**, or the list of scripts currently running on a tab

If there is an extension API (or if it can be done through a Web page-side API with injected JS?), or if we do set up that time accumulation ourselves, it should be pretty easy to add the ctime to [getattr in the code](https://github.com/osnr/TabFS/blob/f875d56b0741099569a2bac82acc1d3a937dcb83/extension/background.js#L437).

I think that you could do some cool subtleties, where the ctime is the time the tab was created and the mtime is the time of last navigation on that tab. (and I guess that would apply to both `by-id/*` and `by-title/*`. maybe `last-focused` has mtime based on the time you focused on that tab?)

**Comment by jtagcat** [comment]
Related: #21

**Comment by vu3rdd** [comment]
There is a `lastAccessed` field in the `Tab` object. I don't see anything else in there related to creation/modification time.

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab
